app/models/user.rb

```diff
@@ -56,4 +56,14 @@ class User < ApplicationRecord
   def matches
     first_matches << second_matches
   end
+
+  def self.from_facebook(user_profile)
```

Comment:
At the moment is this method is correct, but what if in the future your api needs authentication via other social provider? I think that you can change the method name and pass the provider (facebook in this case) as a param